v1/runGame.php

```diff
@@ -11,83 +11,39 @@
-else echo "Argument --height is missing\n";
-
-if (isset($x) && isset($y))
+if ($options->getOption('input'))
 {
-    $gameField = new GameField($x,$y);
-
-    if($options->getOption('type')) {
-
-        switch ($options->getOption('type'))
-        {
-            case "blinker":
-                $gameField->getCellByCoords(3, 2)->life();
-                $gameField->getCellByCoords(3, 3)->life();
-                $gameField->getCellByCoords(3, 4)->life();
-                break;
-
-            case "glider":
-                $gameField->getCellByCoords(2, 1)->life();
-                $gameField->getCellByCoords(3, 2)->life();
-                $gameField->getCellByCoords(3, 3)->life();
-                $gameField->getCellByCoords(2, 3)->life();
-                $gameField->getCellByCoords(1, 3)->life();
-                break;
+    if ($options->getOption("filepath")) $config = array('filePath' => $options->getOption('filepath'));
+    $inputClass = $options->getOption("input")."Input";
+    $input = new $inputClass($config);
 
-            case "lws":
-                $gameField->getCellByCoords(1, 5)->life();
-                $gameField->getCellByCoords(2, 4)->life();
-                $gameField->getCellByCoords(3, 4)->life();
-                $gameField->getCellByCoords(4, 4)->life();
-                $gameField->getCellByCoords(5, 4)->life();
-                $gameField->getCellByCoords(5, 5)->life();
-                $gameField->getCellByCoords(5, 6)->life();
-                $gameField->getCellByCoords(4, 7)->life();
-                $gameField->getCellByCoords(1, 7)->life();
-                break;
+    if ($options->getOption("output"))
+    {
+        $outputClass = $options->getOption("output")."Output";
+        $numCycles = $options->getOption('numCycles');
 
-            default:
-                echo "Wrong type!\n";
-        }
-
-        if($options->getOption('output'))
+        if ($options->getOption("numCycles"))
         {
-            $outputClass = $options->getOption("output")."Output";
-            if($options->getOption('numCycles'))
-            {
-                $numCycles = $options->getOption('numCycles');
-
-                $gameFieldController = new GameFieldController($gameField);
-                $output = new $outputClass();
-                $output->output($gameFieldController, $numCycles);
-            }
-            else echo "Missing --numCycles argument\n";
+            $gameFieldController = new GameFieldController($input->getGameField());
+            $output = new $outputClass();
+            $output->output($gameFieldController, $numCycles);
         }
-        else echo "Missing --output argument\n";
+        else die("Please specify a number of cycles the game should be played");
     }
-    else echo "Missing --type argument\n";
-}
-
-if($options->getOption('help'))
-{
-    $options->showHelp();
+    else die("Please specify an output mode");
 }
+else die("Wrong input");
```

Comment:
"Wrong input" is very less. Please implement the help param and explain all input and output possibilities.
Than you can write here "You didn't define any input possibility. Check help on how to do that!"

if ($options->getOption("help"))
{
echo "YOUR HELP TEXT"
}